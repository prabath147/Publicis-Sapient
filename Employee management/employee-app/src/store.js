import { createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./Component/reducers";
import Thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(Thunk))
);

export default store;
