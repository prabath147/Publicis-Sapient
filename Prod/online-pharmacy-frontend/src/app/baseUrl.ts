export const baseURL = "https://frk5iphcfk.execute-api.us-east-1.amazonaws.com/";
    // "http://localhost:8081";
    
