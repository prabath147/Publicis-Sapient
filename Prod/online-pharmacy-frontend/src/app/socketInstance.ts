
export const socketURL = "https://c8zpvs7fv4.execute-api.us-east-1.amazonaws.com/notify-ws"

// const socket = new SockJS(socketURL);
// export const stompClient = Stomp.over(socket);