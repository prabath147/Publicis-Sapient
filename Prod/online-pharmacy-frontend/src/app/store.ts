import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import StoreReducer from "../components/pages/Admin/storeList/StoreSlice";
import SubscriptionReducer from "../components/pages/AdminSubscriptionPack/SubscriptionvSlice";
import UserReducer from "../components/pages/Auth/Login/UserSlice";
import CartReducer from "../components/pages/Cart/CartSlice";
import ManagerReducer from "../components/pages/Manager/ManagerSlice";
import NotificationReducer from "../components/pages/Open/notificationList/NotificationSlice";
// import OrderSlice from "../components/pages/orderDetail/OrderSlice";
// import OrderListSlice from "../components/pages/orderList/OrderListSlice";

export const store = configureStore({
  reducer: {
    storeL: StoreReducer,
    user: UserReducer,
    notification: NotificationReducer,
    viewsubscription: SubscriptionReducer,
    // addsubscription: AddSubcriptionRe,
    cart: CartReducer,
    // order: OrderSlice,
    // orderL: OrderListSlice,
    manager: ManagerReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
