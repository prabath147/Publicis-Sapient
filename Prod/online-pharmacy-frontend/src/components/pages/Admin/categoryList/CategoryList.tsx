import {
  Button,
  Card,
  Center,
  Container,
  Grid,
  Loader,
  Pagination,
  Space,
  Table,
  TextInput
} from "@mantine/core";
import { useForm } from "@mantine/form";
import { showNotification } from "@mantine/notifications";
import { useEffect, useState } from "react";
import {
  createCategory,
  deleteCategory,
  getCategoryList
} from "./CategoryListAPI";
import { Category } from "./models";

export default function CategoryList() {
  const [activePage, setPage] = useState(1);
  const [pageCount, setPageCount] = useState(1);
  const [loading, setLoading] = useState(true);
  const [categories, setCategories] = useState<Category[]>([]);

  const limit = 10;

  function getCategories() {
    setLoading(true);
    getCategoryList(activePage - 1, limit)
      .then((response) => {
        setPageCount(response.data.totalPages);
        setCategories(response.data.data);
        setLoading(false);

      })
      .catch(() => {
        showNotification({
          color: "red",
          message: "oops, something went wrong",
        });
        setCategories([]);
        setLoading(false);

      })

  }

  useEffect(() => {
    getCategories();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [activePage]);

  const form = useForm<Category>({
    validateInputOnChange: true,
    initialValues: {
      categoryId: -1,
      categoryName: "",
    },
    validate: {
      categoryName: (value: string) =>
        value.length < 2 ? "it should be have a length of 2" : null,
    },
  });

  if (loading)
    return (
      <Center style={{ height: "80%" }}>
        <Loader data-testid="Loading" />
      </Center>
    );

  return (
    <Container>
      <Card>
        <form>
          <Grid>
            <Grid.Col>
              <TextInput
                data-testid="CategoryInput"
                mt={"md"}
                label="Category Name"
                withAsterisk
                {...form.getInputProps("categoryName")}
              />
            </Grid.Col>
            <Grid.Col>
              <Center>
                <Button
                  data-testid="Create-Category-Button"
                  onClick={() => {
                    createCategory(form.values)
                      .then(() => {
                        showNotification({
                          color: "green",
                          message: "Category created successfully.",
                        });
                        getCategories();
                        form.setValues({
                          categoryName: "",
                          categoryId: -1,
                        });
                      })
                      .catch(() => {
                        showNotification({
                          color: "red",
                          message: "Oops, something went wrong.",
                        });
                      });
                  }}
                >
                  Create
                </Button>
              </Center>
            </Grid.Col>
          </Grid>
        </form>
      </Card>
      <Table>
        <thead>
          <tr>
            <th>Id</th>
            <th>Name</th>
            <th>Operations</th>
          </tr>
        </thead>
        <tbody>
          {categories.map((category) => (
            <tr key={category.categoryId}>
              <td>{category.categoryId}</td>
              <td>{category.categoryName}</td>
              <td>
                <Button
                  size="sm"
                  color="red"
                  data-testid={"category-list-" + category.categoryId}
                  onClick={() => {
                    deleteCategory(category.categoryId)
                      .then(() => {
                        getCategories();
                      })
                      .catch(() => {
                        showNotification({
                          color: "red",
                          message: "Oops! Something went wrong",
                        });
                      });
                  }}
                >
                  Delete
                </Button>
              </td>
            </tr>
          ))}
        </tbody>
      </Table>
      <Space h="xl" />

      <Pagination
        position="center"
        page={activePage}
        onChange={setPage}
        total={pageCount}
      />
    </Container>
  );
}
