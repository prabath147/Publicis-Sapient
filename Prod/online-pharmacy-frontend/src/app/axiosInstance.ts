import axios, { AxiosRequestConfig } from "axios";
import LocalUserService from "../services/LocalUserService";
import TokenService from "../services/TokenService";
import { baseURL } from "./baseUrl";

const axiosInstance = axios.create({
  baseURL: baseURL,
  timeout: 5000,
  headers: {
    "Content-Type": "application/json",
    accept: "application/json",
  },
});

axiosInstance.interceptors.request.use(
  (config: AxiosRequestConfig) => {
    const token = TokenService.getLocalAccessToken();
    if (token && config.headers !== undefined) {
      config.headers["Authorization"] = "Bearer " + token; // for Spring Boot back-end
      // config.headers["x-access-token"] = token; // for Node.js Express back-end
      config.headers["ngrok-skip-browser-warning"] = "";
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

axiosInstance.interceptors.response.use(
  (res) => {
    return res;
  },
  async (err) => {
    const originalConfig = err.config;
    if (originalConfig.url !== "api/auth/signin" && err.response) {
      // Access Token was expired
      if (err.response.status === 401 && !originalConfig._retry) {
        originalConfig._retry = true;
        try {
          const rs = await axiosInstance.post("api/auth/refresh-token", {
            refreshToken: TokenService.getLocalRefreshToken(),
          });
          const { accessToken } = rs.data;
          TokenService.updateLocalAccessToken(accessToken);
          return axiosInstance(originalConfig);
        } catch (_error) {
          TokenService.removeLocalAccessToken()
          TokenService.removeLocalRefreshToken()
          LocalUserService.removeLocalUser()
          // window.location.reload();
          return Promise.reject(_error);
        }
      }
    }
    return Promise.reject(err);
  }
);
export default axiosInstance;
