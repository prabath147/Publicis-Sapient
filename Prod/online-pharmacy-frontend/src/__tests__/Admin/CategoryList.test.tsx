import { act, fireEvent, render, waitForElementToBeRemoved } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { store } from "../../app/store";
import CategoryList from "../../components/pages/Admin/categoryList/CategoryList";
import mockAxios from "../../__mocks__/axios";

const MockCategoryL = () => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <CategoryList />
      </BrowserRouter>
    </Provider>
  );
};

it("component created", () => {
  const component = renderer.create(<MockCategoryL />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

const category = {
  data: [{ categoryId: 10000042, categoryName: "category0" }],
  pageNumber: 0,
  pageSize: 10,
  totalRecords: 1,
  totalPages: 1,
  isLastPage: true,
};
describe("Category list", () => {
  afterEach(() => {
    jest.resetAllMocks();
  });


  it("component created without data", async () => {
    mockAxios.get.mockImplementation(() => Promise.resolve({
      data: category
    }));

    const { asFragment, getByTestId } = render(<MockCategoryL />);
    await waitForElementToBeRemoved(getByTestId(/Loading/i));

    expect(asFragment()).toMatchSnapshot();
  });

  it("component created with fail event", async () => {
    mockAxios.get.mockImplementation(() => Promise.reject());

    const { asFragment, getByTestId } = render(<MockCategoryL />);

    await waitForElementToBeRemoved(getByTestId(/Loading/i));

    expect(asFragment()).toMatchSnapshot();
  });

  it("Creating category with incorrect data", async () => {
    mockAxios.get.mockImplementation(() =>
      Promise.resolve({ data: { data: [] } })
    );
    mockAxios.post.mockImplementation(() =>
      Promise.resolve()
    );
    const { asFragment, getByTestId } = render(<MockCategoryL />);

    await waitForElementToBeRemoved(getByTestId(/Loading/i));

    await act(async () => {
      fireEvent.change(getByTestId(/CategoryInput/i), {
        target: { value: "c" }
      })
    })

    await act(async () => {
      fireEvent.click(getByTestId(/Create-Category-Button/i));
    })

    // await user
    //   .click(createBtn)
    //   .then(mockAxios.post.mockImplementationOnce(() => Promise.reject()));
  });

  it("Creating category with correct data", async () => {
    mockAxios.get.mockImplementation(() =>
      Promise.resolve({ data: { data: [] } })
    );
    mockAxios.post.mockImplementation(() =>
      Promise.resolve()
    );
    const { asFragment, getByTestId } = render(<MockCategoryL />);

    await waitForElementToBeRemoved(getByTestId(/Loading/i));

    await act(async () => {
      fireEvent.change(getByTestId(/CategoryInput/i), {
        target: { value: "c" }
      })
    })

    await act(async () => {
      fireEvent.click(getByTestId(/Create-Category-Button/i));
    })

    // await act(async () => {
    //   fireEvent.click(getByTestId(/Create-Category-Button/i))
    // .then(
    //   mockAxios.post.mockImplementationOnce(() =>
    //     Promise.resolve({
    //       categoryId: 10000226,
    //       categoryName: " category700",
    //     })
    //   )
    // )
    // .then(
    //   mockAxios.get.mockImplementationOnce(() =>
    //     Promise.resolve({
    //       data: {
    //         data: [{ categoryId: 10000226, categoryName: "category700" }],
    //         pageNumber: 0,
    //         pageSize: 10,
    //         totalRecords: 11,
    //         totalPages: 2,
    //         isLastPage: false,
    //       },
    //     })
    //   )
    // );
    // })

  });
});


