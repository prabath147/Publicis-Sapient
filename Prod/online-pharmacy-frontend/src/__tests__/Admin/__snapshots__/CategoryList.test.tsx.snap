// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category list component created with fail event 1`] = `
<DocumentFragment>
  <div
    class="mantine-Container-root mantine-15pjuqq"
  >
    <div
      class="mantine-Paper-root mantine-Card-root mantine-xoj88x"
    >
      <form>
        <div
          class="mantine-Grid-root mantine-nsdj"
        >
          <div
            class="mantine-Col-root mantine-lungan"
          >
            <div
              class="mantine-InputWrapper-root mantine-TextInput-root mantine-1l2unfb"
            >
              <label
                class="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
                for="mantine-r1"
                id="mantine-r1-label"
              >
                Category Name
                <span
                  aria-hidden="true"
                  class="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
                >
                   *
                </span>
              </label>
              <div
                class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
              >
                <input
                  aria-invalid="false"
                  class="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
                  data-testid="CategoryInput"
                  id="mantine-r1"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="mantine-Col-root mantine-lungan"
          >
            <div
              class="mantine-8od8ev"
            >
              <button
                class="mantine-UnstyledButton-root mantine-Button-root mantine-8nr514"
                data-button="true"
                data-testid="Create-Category-Button"
                type="button"
              >
                <div
                  class="mantine-3xbgk5 mantine-Button-inner"
                >
                  <span
                    class="mantine-qo1k2 mantine-Button-label"
                  >
                    Create
                  </span>
                </div>
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <table
      class="mantine-Table-root mantine-b146xb"
    >
      <thead>
        <tr>
          <th>
            Id
          </th>
          <th>
            Name
          </th>
          <th>
            Operations
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
    <div
      class="mantine-1eabysl"
    />
    <div
      class="mantine-Group-root mantine-1q3od06"
      role="navigation"
    >
      <button
        aria-disabled="true"
        class="mantine-Group-child mantine-Pagination-item mantine-1bnfkxz"
        disabled=""
        type="button"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.219 8l3.3 3.3-.943.943L5.333 8l4.243-4.243.943.943-3.3 3.3z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        aria-current="page"
        class="mantine-Group-child mantine-Pagination-item mantine-1bnfkxz"
        data-active="true"
        tabindex="0"
        type="button"
      >
        1
      </button>
      <button
        aria-disabled="true"
        class="mantine-Group-child mantine-Pagination-item mantine-1bnfkxz"
        disabled=""
        type="button"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.781 8l-3.3-3.3.943-.943L10.667 8l-4.243 4.243-.943-.943 3.3-3.3z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Category list component created without data 1`] = `
<DocumentFragment>
  <div
    class="mantine-Container-root mantine-15pjuqq"
  >
    <div
      class="mantine-Paper-root mantine-Card-root mantine-xoj88x"
    >
      <form>
        <div
          class="mantine-Grid-root mantine-nsdj"
        >
          <div
            class="mantine-Col-root mantine-lungan"
          >
            <div
              class="mantine-InputWrapper-root mantine-TextInput-root mantine-1l2unfb"
            >
              <label
                class="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
                for="mantine-r0"
                id="mantine-r0-label"
              >
                Category Name
                <span
                  aria-hidden="true"
                  class="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
                >
                   *
                </span>
              </label>
              <div
                class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
              >
                <input
                  aria-invalid="false"
                  class="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
                  data-testid="CategoryInput"
                  id="mantine-r0"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="mantine-Col-root mantine-lungan"
          >
            <div
              class="mantine-8od8ev"
            >
              <button
                class="mantine-UnstyledButton-root mantine-Button-root mantine-8nr514"
                data-button="true"
                data-testid="Create-Category-Button"
                type="button"
              >
                <div
                  class="mantine-3xbgk5 mantine-Button-inner"
                >
                  <span
                    class="mantine-qo1k2 mantine-Button-label"
                  >
                    Create
                  </span>
                </div>
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <table
      class="mantine-Table-root mantine-b146xb"
    >
      <thead>
        <tr>
          <th>
            Id
          </th>
          <th>
            Name
          </th>
          <th>
            Operations
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            10000042
          </td>
          <td>
            category0
          </td>
          <td>
            <button
              class="mantine-UnstyledButton-root mantine-Button-root mantine-ofjsjw"
              data-button="true"
              data-testid="category-list-10000042"
              type="button"
            >
              <div
                class="mantine-3xbgk5 mantine-Button-inner"
              >
                <span
                  class="mantine-qo1k2 mantine-Button-label"
                >
                  Delete
                </span>
              </div>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="mantine-1eabysl"
    />
    <div
      class="mantine-Group-root mantine-1q3od06"
      role="navigation"
    >
      <button
        aria-disabled="true"
        class="mantine-Group-child mantine-Pagination-item mantine-1bnfkxz"
        disabled=""
        type="button"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.219 8l3.3 3.3-.943.943L5.333 8l4.243-4.243.943.943-3.3 3.3z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        aria-current="page"
        class="mantine-Group-child mantine-Pagination-item mantine-1bnfkxz"
        data-active="true"
        tabindex="0"
        type="button"
      >
        1
      </button>
      <button
        aria-disabled="true"
        class="mantine-Group-child mantine-Pagination-item mantine-1bnfkxz"
        disabled=""
        type="button"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.781 8l-3.3-3.3.943-.943L10.667 8l-4.243 4.243-.943-.943 3.3-3.3z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`component created 1`] = `
<div
  className="mantine-8od8ev"
  style={
    Object {
      "height": "80%",
    }
  }
>
  <svg
    className="mantine-1avyp1d"
    data-testid="Loading"
    height="36px"
    role="presentation"
    stroke="#228be6"
    viewBox="0 0 38 38"
    width="36px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fillRule="evenodd"
    >
      <g
        strokeWidth="5"
        transform="translate(2.5 2.5)"
      >
        <circle
          cx="16"
          cy="16"
          r="16"
          strokeOpacity=".5"
        />
        <path
          d="M32 16c0-9.94-8.06-16-16-16"
        >
          <animateTransform
            attributeName="transform"
            dur="1s"
            from="0 16 16"
            repeatCount="indefinite"
            to="360 16 16"
            type="rotate"
          />
        </path>
      </g>
    </g>
  </svg>
</div>
`;
